api/src/main/java/org/xnio/sasl/SaslStrength.java

```diff
@@ -28,13 +30,47 @@
     /**
      * Specify low cipher strength.
      */
-    LOW,
+    LOW("low"),
     /**
      * Specify medium cipher strength.
      */
-    MEDIUM,
+    MEDIUM("medium"),
     /**
      * Specify high cipher strength.
      */
-    HIGH,
+    HIGH("high"),
+    ;
+
+    private final String toString;
+
+    SaslStrength(String toString) {
+        this.toString = toString;
+    }
+
+    /**
+     * Get the SASL Strength level for the given string.
+     *
+     * @param name the Strength level
+     * @return the Strength value
+     */
+    public static SaslStrength fromString(String name) {
```

Comment:
When would this be used?

Reply:
I've created this PR because this processing is currently done in wildfly-core [1]. IMO the cleaner approach would be to have the string representation directly in the enum to avoid possible change at several places. The same approach is taken in SaslQop enum [2]

[1] https://github.com/wildfly/wildfly-core/blob/master/remoting/subsystem/src/main/java/org/jboss/as/remoting/ConnectorUtils.java#L186
[2] https://github.com/xnio/xnio/blob/3.x/api/src/main/java/org/xnio/sasl/SaslQop.java